common/nymsphinx/chunking/src/reconstruction.rs

```diff
@@ -296,6 +307,25 @@ impl MessageReconstructor {
     pub fn recover_fragment(&self, fragment_data: Vec<u8>) -> Result<Fragment, ChunkingError> {
         Fragment::try_from_bytes(&fragment_data)
     }
+
+    pub fn cleanup_stale_buffers(&mut self) {
+        trace!("Cleaning up stale buffers");
+        let now = Instant::now();
+        self.reconstructed_sets.retain(|_, set_buf| {
```

Comment:
I wonder, and I know I'm contradicting myself, but should we be calling `cleanup_stale_buffers` on **every** fragment? reason being is that cost of `.retain` is non-negligible as it has to iterate through whole thing every time. though maybe if we don't expect a lot of entries there it's fine. 

perhaps we could keep internal counter of fragments received and call `cleanup_stale_buffers` if either `received % 1000 == 0 `( or whatever constant) or `last_cleaned_up < threshold`?